dojo/jira_link/views.py

```diff
@@ -125,23 +126,68 @@ def webhook(request, secret=None):
+                            "32x32":"https://www.gravatar.com/avatar/9637bfb970eff6176357df615f548f1c?d=mm&s=32"
+                        },
+                        "displayName":"Defect Dojo",
+                        "active":true,
+                        "timeZone":"Europe/Amsterdam"
+                    },
+                    "body":"(Valentijn Scholten):test4",
+                    "updateAuthor":{
+                        "self":"https://jira.host.com/rest/api/2/user?username=defect.dojo",
+                        "name":"defect.dojo",
+                        "key":"defect.dojo",
+                        "avatarUrls":{
+                            "48x48":"https://www.gravatar.com/avatar/9637bfb970eff6176357df615f548f1c?d=mm&s=48",
+                            "24x24""https://www.gravatar.com/avatar/9637bfb970eff6176357df615f548f1c?d=mm&s=24",
+                            "16x16":"https://www.gravatar.com/avatar/9637bfb970eff6176357df615f548f1c?d=mm&s=16",
+                            "32x32":"https://www.gravatar.com/avatar/9637bfb970eff6176357df615f548f1c?d=mm&s=32"
+                        },
+                        "displayName":"Defect Dojo",
+                        "active":true,
+                        "timeZone":"Europe/Amsterdam"
+                    },
+                    "created":"2021-01-10T10:12:47.824+0100",
+                    "updated":"2021-01-10T10:12:47.824+0100"
+                }
+                }
+            """
+
             comment_text = parsed['comment']['body']
-            commentor = parsed['comment']['updateAuthor']['displayName']
+            commentor = parsed['comment']['updateAuthor']['key']
+            commentor_display_name = parsed['comment']['updateAuthor']['displayName']
             # example: body['comment']['self'] = "http://www.testjira.com/jira_under_a_path/rest/api/2/issue/666/comment/456843"
             jid = parsed['comment']['self'].split('/')[-3]
             jissue = get_object_or_404(JIRA_Issue, jira_id=jid)
             logger.debug('jissue: %s', vars(jissue))
             if jissue.finding:
-                logger.debug('finding: %s', vars(jissue.finding))
+                # logger.debug('finding: %s', vars(jissue.finding))
                 jira_usernames = JIRA_Instance.objects.values_list('username', flat=True)
                 for jira_userid in jira_usernames:
-                    if jira_userid.lower() in commentor.lower():
-                        logger.info('skipping incoming JIRA comment as the user id of the comment mathces the JIRA user in Defect Dojo')
+                    # logger.debug('incoming username: %s jira config username: %s', commentor.lower(), jira_userid.lower())
+                    if jira_userid.lower() == commentor.lower():
+                        logger.debug('skipping incoming JIRA comment as the user id of the comment in JIRA (%s) matches the JIRA username in Defect Dojo (%s)', commentor.lower(), jira_userid.lower())
```

Comment:
I am thinking maybe at times, there could be corner cases where the jira userid and the commentor name could match, and possibly, a legit comment would not make it to defectdojo?

How about making this an `logger.info`?

Reply:
It's almost impossible to match by accident. In that case the defect.dojo user in JIRA is used to place a comment directly into JIRA?
This log line will happen *for every* note posted to dojo where a JIRA issue is attached to the finding, so it should be not logged at all or only at debug level.
If we worry about comments being missed we should improve the check and look for matching timestamps of the note in dojo and jira etc.